src/python/WMCore/MicroService/Unified/MSOutput.py

```diff
@@ -288,34 +288,30 @@ def makeSubscriptions(self, workflow):
                 ddmReqList = []
                 try:
                     if workflow['isRelVal']:
-                        for dMap in workflow['destinationOutputMap']:
-                            try:
-                                ddmRequest = DDMReqTemplate('copy',
-                                                            item=dMap['datasets'],
-                                                            n=workflow['numberOfCopies'],
-                                                            site=dMap['destination'],
-                                                            group='RelVal')
-                            except KeyError as ex:
-                                # NOTE:
-                                #    If we get to here it is most probably because the 'site'
-                                #    mandatory field to the DDM request is missing (due to an
-                                #    'ALCARECO' dataset or similar). Since this is expected
-                                #    to happen a lot, we'd better just log a warning and continue
-                                msg = "Could not create DDMReq for Workflow: {}".format(workflow['RequestName'])
-                                msg += "Error: {}".format(ex)
-                                self.logger.warning(msg)
-                                continue
-                            ddmReqList.append(ddmRequest)
+                        group = 'RelVal'
                     else:
-                        # FIXME:
-                        #    We need to create the campaignMap and use it for
-                        #    creating the requests for the nonRelVal workflows
-                        #    it should be also the case when we migrate to Rucio
-                        ddmRequest = DDMReqTemplate('copy',
-                                                    item=workflow['OutputDatasets'],
-                                                    n=workflow['numberOfCopies'],
-                                                    site=workflow['destination'])
+                        group = 'DataOps'
+
+                    for dMap in workflow['destinationOutputMap']:
+                        try:
+                            ddmRequest = DDMReqTemplate('copy',
+                                                        item=dMap['datasets'],
+                                                        n=workflow['numberOfCopies'],
+                                                        site=dMap['destination'],
+                                                        group=group)
+                        except KeyError as ex:
+                            # NOTE:
+                            #    If we get to here it is most probably because the 'site'
```

Comment:
Wait, don't we make a DDM request without any destination for standard workflows? Which means, `site` will only have a non-None value for RelVal samples (that match the data tier)

Reply:
No, For standard workflows, the fact that we do or do not define the `site` here or in the the producer thread, does not change anything - the Wildcard list of `['T1_*', 'T2_*']` is set by default from the DDM module anyway [1]. But when it comes to `RelVal`s we can get into the trouble of a workflow with a zeroed site list because of: [2]. This check does not come with this PR I was experiencing that error back before I start working on unifying the submission logic for Relvals and NonRelvals,  and that's why it is put there. 


[1]
https://github.com/dmwm/WMCore/blob/6c7032b85841a535247026ba70b48686fccd2838/src/python/WMCore/Services/DDM/DDM.py#L42

[2]
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/MicroService/Unified/MSOutput.py#L668
https://github.com/dmwm/WMCore/blob/6c7032b85841a535247026ba70b48686fccd2838/src/python/WMCore/MicroService/Unified/MSOutput.py#L676
https://github.com/dmwm/WMCore/blob/6c7032b85841a535247026ba70b48686fccd2838/src/python/WMCore/MicroService/Unified/MSOutput.py#L678

Reply:
Okay, I see you have also updated the wiki and that this is actually the default in Dynamo for cases that a destination site does not get provided by the client. We might have to touch this default in the future, when going to Rucio, but we can likely live it like that for now.

> But when it comes to RelVals we can get into the trouble of a workflow with a zeroed site list because of: [2]

I'm not sure I understood it. Are you saying that same Relval output datasets do not have any location, because it's by design that we do not transfer some of them? What is the problem with that?